modules/consensus/accept.go

```diff
@@ -207,78 +195,110 @@ func (cs *ConsensusSet) addBlockToTree(b types.Block) (ce changeEntry, err error
 // This method is typically only be used when there would otherwise be multiple
 // consecutive calls to AcceptBlock with each successive call accepting the
 // child block of the previous call.
-func (cs *ConsensusSet) managedAcceptBlock(b types.Block) error {
-	// Grab a lock on the consensus set. Lock is demoted later in the function,
-	// failure to unlock before returning an error will cause a deadlock.
+func (cs *ConsensusSet) managedAcceptBlocks(blocks []types.Block) (bool, error) {
```

Comment:
Looks like we need to reconsider  the interplay between the bool and the error - if we are accepting a whole batch of blocks, the blocks could individually have different errors, and some of them may extend the chain while others do not.

In a reorg situation, you could end up with a bunch of blocks that don't extend the chain followed by only one that does. I'm not sure how that plays with the rest of the code, but my initial leaning is to return an `[]error` instead of a `bool, error`, that way you can return the errors for each block individually, and then handle them as you would have if you had submitted the blocks in a loop.